gax-bom/pom.xml

```diff
@@ -7,6 +7,13 @@
   <packaging>pom</packaging>
   <name>GAX (Google Api eXtensions) for Java (BOM)</name>
   <description>Google Api eXtensions for Java (BOM)</description>
+
```

Comment:
What about making the root pom the parent which is already extending `shared-config`? That seems more aligned with what we are doing in other repos including [google-cloud-java](https://github.com/googleapis/google-cloud-java/blob/7e32e278c2fae7b07a9040e74afaf3cf7417642e/gapic-libraries-bom/pom.xml#L15)

Reply:
I was debating the two approaches and remembered a conversation from Tomo in the past. Having the `bom` inherit shared-configs would be cleaner as it gives us a simple `<dependencyManagement>` section:
```
<dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax</artifactId>
        <version>2.19.5</version>
      </dependency>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax</artifactId>
        <version>2.19.5</version>
        <classifier>testlib</classifier>
      </dependency>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax-grpc</artifactId>
        <version>2.19.5</version>
      </dependency>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax-grpc</artifactId>
        <version>2.19.5</version>
        <classifier>testlib</classifier>
      </dependency>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax-httpjson</artifactId>
        <version>0.104.5</version>
      </dependency>
      <dependency>
        <groupId>com.google.api</groupId>
        <artifactId>gax-httpjson</artifactId>
        <version>0.104.5</version>
        <classifier>testlib</classifier>
      </dependency>
      <dependency>
        <groupId>com.google.auto.value</groupId>
        <artifactId>auto-value-annotations</artifactId>
        <version>1.10</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
```
whereas having it inherit the root pom would merge all the `<dependencyManagement>` sections from all parents poms (https://github.com/googleapis/gax-java/blob/main/pom.xml#L54-L176). We would accidentally specify the versions for some additional dependencies and that *may* lead to some dependency issues in the future.

IIRC, this is the same approach we take in the monorepo. We have two "parent" poms (jar-parent and pom-parent):
- Jar-Parent is the parent for all the non-BOM modules (Dependencies sections specified for Jars only: https://github.com/googleapis/google-cloud-java/blob/7e32e278c2fae7b07a9040e74afaf3cf7417642e/google-cloud-jar-parent/pom.xml#L25-L122
- Pom-Parent is the parent for the BOMs (Jar-Parent's parent is the Pom-Parent)